core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -188,7 +204,14 @@ private ScriptEngineService getScriptEngineServiceForFileExt(String fileExtensio
         return scriptEngineService;
     }
 
-
+    void setMaxCompilationsPerMinute(Integer newMaxPerMinute) {
+        if (newMaxPerMinute != null) {
```

Comment:
I'm actually going to remove this null check as it's not needed, the settings infra does all the checking